app/controllers/users_controller.rb

```diff
@@ -19,21 +27,66 @@ def create
     end
   end
 
+  def edit
+    @user = User.find(require_id)
+  end
+
+  def update
+    @user = User.find(require_id)
+    if @user.update(user_params)
+      flash[:success] = I18n.t 'users.edit.success'
+      redirect_to @user
+    else
+      flash[:error] = I18n.t 'users.edit.error'
+      render 'edit'
+    end
+  end
+
   def destroy
-    log_out
-    redirect_to root_url
+    User.find(require_id).destroy
```

Comment:
`User.find(require_id)` is used in `edit`, `update`, `destroy`. Seems like a candidate for a `before_action`. This would also allow you respond uniformly in case no such user is found, by ex: flashing an error instead of the `undefined method 'destroy' for nil` exception that you would get in this case

Reply:
Also `destroy` returns `false` if it fails (which is not being handled here and would be treated as a success), consider handling failure case explicitly (preferable) or using `destroy!`